cmd/kubernetes/kubernetes_remove.go

```diff
@@ -52,53 +62,112 @@ var kubernetesRemoveCmd = &cobra.Command{
 				}
 			}
 			kuberneteList = append(kuberneteList, utility.ObjecteList{ID: kubernetesCluster.ID, Name: kubernetesCluster.Name})
-		} else {
-			for _, v := range args {
-				kubernetesCluster, err := client.FindKubernetesCluster(v)
-				if err == nil {
-					kuberneteList = append(kuberneteList, utility.ObjecteList{ID: kubernetesCluster.ID, Name: kubernetesCluster.Name})
-				}
-			}
 		}
 
+		// Collect names of the clusters to be deleted for confirmation message
 		kubernetesNameList := []string{}
 		for _, v := range kuberneteList {
 			kubernetesNameList = append(kubernetesNameList, v.Name)
 		}
 
-		var volsNameList []string
-		for _, v := range kuberneteList {
-			vols, err := client.ListVolumesForCluster(v.ID)
-			if err != nil {
-				utility.Error("error getting the list of dangling volumes: %s", err)
-				os.Exit(1)
-			}
-			for _, v := range vols {
-				volsNameList = append(volsNameList, v.Name)
-			}
-			if vols != nil {
-				utility.YellowConfirm("There are volume(s) attached to this cluster. Consider deleting or detaching these before deleting the cluster:\n%s\n", utility.Green(strings.Join(volsNameList, "\n")))
-			}
-		}
-
+		// Confirm the deletion with the user
 		if utility.UserConfirmedDeletion(fmt.Sprintf("Kubernetes %s", pluralize.Pluralize(len(kuberneteList), "cluster")), common.DefaultYes, strings.Join(kubernetesNameList, ", ")) {
-
 			for _, v := range kuberneteList {
-				_, err = client.DeleteKubernetesCluster(v.ID)
+
+				// Delete the Kubernetes cluster
+				_, err := client.DeleteKubernetesCluster(v.ID)
 				if err != nil {
 					utility.Error("error deleting the kubernetes cluster: %s", err)
 					os.Exit(1)
 				}
+
+				/* Poll for the deletion status
```

Comment:
we should fix this on API side rather than client side so the clients don't have to poll like this.. cc : @uzaxirr

Reply:
Please correct me if I am wrong @RealHarshThakur  @uzaxirr  @fernando-villalba 

I believe the current API logic is appropriate. The API call should end after triggering the deletion of the cluster, allowing the client-side polling mechanism to take over. This approach ensures that the API call is quick and responsive. Typically, the polling mechanism requires only 1-2 attempts with a good internet connection and up to 3 attempts with a weaker connection to confirm the deletion status.

If we move the polling logic to the API side, the API call might take longer to complete and could potentially result in timeouts or errors, especially under poor network conditions. This could negatively impact user experience and increase the complexity of the API.

I look forward to your thoughts and feedback. Please let me know what I can improve.